ceramic-ahash/src/main.rs

```diff
@@ -0,0 +1,161 @@
+use multihash::{Hasher, Sha2_256};
+use std::fmt::{Debug, Display, Error, Formatter};
+
+pub struct Ahash([u32; 8]);
+impl std::ops::Add for Ahash {
+    type Output = Self;
+    fn add(self, rhs: Self) -> Self::Output {
+        Self([
+            self.0[0].wrapping_add(rhs.0[0]),
+            self.0[1].wrapping_add(rhs.0[1]),
+            self.0[2].wrapping_add(rhs.0[2]),
+            self.0[3].wrapping_add(rhs.0[3]),
+            self.0[4].wrapping_add(rhs.0[4]),
+            self.0[5].wrapping_add(rhs.0[5]),
+            self.0[6].wrapping_add(rhs.0[6]),
+            self.0[7].wrapping_add(rhs.0[7]),
+        ])
+    }
+}
+
+impl Ahash {
+    fn sha256_digest(s: &str) -> [u8; 32] {
+        let mut hasher = Sha2_256::default();
+        hasher.update(s.as_bytes());
+        // sha256 is 32 bytes safe to unwrap to [u8; 32]
+        hasher.finalize().try_into().unwrap()
+    }
+
+    pub fn digest(key: &str) -> Ahash {
+        let d = Ahash::sha256_digest(key);
+        Ahash([
+            // 4 byte slices safe to unwrap to [u8; 4]
+            u32::from_le_bytes(d[0..4].try_into().unwrap()),
+            u32::from_le_bytes(d[4..8].try_into().unwrap()),
+            u32::from_le_bytes(d[8..12].try_into().unwrap()),
+            u32::from_le_bytes(d[12..16].try_into().unwrap()),
+            u32::from_le_bytes(d[16..20].try_into().unwrap()),
+            u32::from_le_bytes(d[20..24].try_into().unwrap()),
+            u32::from_le_bytes(d[24..28].try_into().unwrap()),
+            u32::from_le_bytes(d[28..32].try_into().unwrap()),
+        ])
+    }
+
+    pub fn digest_many<'a, I>(keys: I) -> Ahash
+    where
+        I: Iterator<Item = &'a str>,
+    {
+        let mut total = Ahash::identity();
+        for key in keys {
+            total = total + Ahash::digest(key)
+        }
+        total
+    }
+
+    pub fn to_hex(&self) -> String {
+        vec![
+            hex::encode(u32::to_le_bytes(self.0[0])),
+            hex::encode(u32::to_le_bytes(self.0[1])),
+            hex::encode(u32::to_le_bytes(self.0[2])),
+            hex::encode(u32::to_le_bytes(self.0[3])),
+            hex::encode(u32::to_le_bytes(self.0[4])),
+            hex::encode(u32::to_le_bytes(self.0[5])),
+            hex::encode(u32::to_le_bytes(self.0[6])),
+            hex::encode(u32::to_le_bytes(self.0[7])),
+        ]
+        .join("")
+        .to_uppercase()
+    }
+
+    pub fn identity() -> Ahash {
```

Comment:
Rust has a `Default` trait which is similar in concept. Here, perhaps implement the `Default` trait to make this behavior more generally useful?  You should be able to derive it `#[derive(Default)]`